presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueryFramework.java

```diff
@@ -162,17 +162,13 @@ protected void assertQueryFails(@Language("SQL") String sql, @Language("RegExp")
 
     protected void assertQueryFails(Session session, @Language("SQL") String sql, @Language("RegExp") String expectedMessageRegExp)
     {
-        queryRunner.getExclusiveLock().lock();
```

Comment:
I see two possibilities here:

* there is no concurrency, and thus no locking is needed -- then locking should be removed from other methods too
* there might be some concurrency -- then at least a shared lock needs to stay here